### PR TITLE
[wrangler] Fix local mode the registry server closed

### LINKED ISSUE #1920: 🐛 BUG: fetching bound registered workers in local mode fails with timeout
### What version of `Wrangler` are you using?

2.1.6

### What operating system are you using?

M1 Pro macOS 12.6

### Describe the Bug

When running `wrangler pages dev`, after some delay (presumably the timeout mentioned in the error) this error is being output several times per second until the server is stopped.

```
▲ [WARNING] Failed to get worker definitions TypeError: fetch failed

      at Object.processResponse
  (/Users/andre/code/wrangler-bug/node_modules/wrangler/wrangler-dist/cli.js:8846:27)
      at /Users/andre/code/wrangler-bug/node_modules/wrangler/wrangler-dist/cli.js:9178:42
      at node:internal/process/task_queues:140:7
      at AsyncResource.runInAsyncScope (node:async_hooks:203:9)
      at AsyncResource.runMicrotask (node:internal/process/task_queues:137:8)
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
    cause: ConnectTimeoutError: Connect Timeout Error
        at onConnectTimeout
  (/Users/andre/code/wrangler-bug/node_modules/wrangler/wrangler-dist/cli.js:2870:29)
        at /Users/andre/code/wrangler-bug/node_modules/wrangler/wrangler-dist/cli.js:2828:51
        at Immediate._onImmediate
  (/Users/andre/code/wrangler-bug/node_modules/wrangler/wrangler-dist/cli.js:2859:13)
        at process.processImmediate (node:internal/timers:471:21) {
      code: 'UND_ERR_CONNECT_TIMEOUT'
    }
  }
```

### steps to reproduce:

 - `npx create-remix@latest` 
     - name
     - Just The Basics
     - Cloudflare Pages
     - Typescript
     - Y
 - `npm run dev` 
 - wait 10-15 seconds.

Note: I do not have wrangler installed globally, and have tried running this same command after running `npx wrangler login` and the error is still output.

### PR DESCRIPTION
**Pull request by oustn**
Fixes #1920 #2959.

**What this PR solves / how to test:**

Solve dev mode the local registry server be closed and throw error: '[WARNING] Failed to get worker definitions TypeError: fetch failed'

**Author has included the following, where applicable:**

- [x] Tests
- [x] Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))

**Reviewer has performed the following, where applicable:**

- [x] Checked for inclusion of relevant tests
- [x] Checked for inclusion of a relevant changeset
- [x] Checked for creation of associated docs updates
- [x] Manually pulled down the changes and spot-tested

### CODE REVIEW

**Comment by petebacondarwin** [comment]
Thanks for providing this fix @oustn.
Do you think you might be able to create a test to demonstrate the fix and to prevent regressions? Probably inside the `unstableDev.test.ts` file.
Also could you add a link to `Fixes #1920` in the changeset and provide a little bit more information there?

File: .changeset/gentle-yaks-repeat.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"wrangler": major
```
Comment (threepointone) [inline .changeset/gentle-yaks-repeat.md:2]: this should be `patch`

Comment (rozenmd) [inline .changeset/gentle-yaks-repeat.md:2]:
```suggestion
"wrangler": patch
```

File: .changeset/gentle-yaks-repeat.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+---
+
+Fix: fix local registry server closed
```
Comment (rozenmd) [inline .changeset/gentle-yaks-repeat.md:5]:
```suggestion
Fix: fix local registry server closed

Closes #1920 
```

**Comment by oustn** [comment]
> Thanks for providing this fix @oustn. Do you think you might be able to create a test to demonstrate the fix and to prevent regressions? Probably inside the `unstableDev.test.ts` file. Also could you add a link to `Fixes #1920` in the changeset and provide a little bit more information there?

Yes, add a simple test

**Comment by rozenmd** [comment]
@cloudflare/wrangler I have pulled down this change and locally tested it, it resolves the issue (finally).

**Review by JacobMGEvans — APPROVED** [review]

**Comment by wighawag** [comment]
I tested with `pnpm add -D https://prerelease-registry.devprod.cloudflare.dev/workers-sdk/prs/3048/npm-package-wrangler-3048`

And I am still getting this issue whenever I change the code while `wrangler dev` is executing

**Pull request merged**